### ISSUE TITLE
1.18.6 .NET 7 - Critical Error's System.Exception: Don't know how to instantiate block of class 'BlockFirebrickKilnFlue' did you forget to register a mapping?

### ISSUE THREAD

**Issue by Diometus** [anegostudios/VintageStory-Issues#2881]
### Game Version

v1.18.6 .NET 7

### Platform

Windows

### Modded

Modded

### SP/MP

Multiplayer

### Description

Getting a critical error when i try to use this in multiplayer. If I disable the mod Volumetric Shading (Updated), I no longer crash at the client loading phase of entering the world/server. However, I can play single-player with the same server mod list with Volumetric Shading (Updated) turned on and no crash so IDK why it's doing this. Especially since there are similar bugs that were reported in the past such as BlockEntityBerryBush. Do I report this here or do I just report it to Volumetric Shading (Updated) mod author? At this point, I've submitted to both the Vintage Story devs and the mod author. 

### How to reproduce

1. Get the mods I am using on the game version I am using.
2. Create a singleplayer world with all the mods in my mod list enabled.
3. Play around with Volumetric Shading (Updated) settings in game and save world.
4. Login to your already-made character that is a Clocksmith/Clockworker on Rogue House server that runs a v1.18.6 server.
5. You should get the error. If you do not, why are you not getting it but I am when playing on that server? 

### Logs

```Running on 64 bit Windows with 65442 MB RAM
Game Version: .net7 Experimental build - v1.18.6 (Unstable)
7/2/2023 9:46:56 AM: Critical error occurred
Loaded Mods: aculinaryartillery@1.0.12, ancienttools@1.5.7, autowalk@1.0.2, chiseltools@1.8.4, instruments@1.2.1, mannequinstand@1.0.1, meteoricexpansion@1.2.4, leadroof@1.3.4, primitivesurvival@3.2.4, spyglass@0.4.5, tprunes@1.0.8, temperaturescales@1.0.2, temporalstormgear@1.0.1, volumetricshadingupdated@0.7.2-pre.1, game@1.18.6, wildcraft@1.6.6, altmapiconrenderer@1.1.1, betterfirepit@1.1.1, bodyheatbar@1.0.1, calendar@1.0.0, canmarket@0.3.9, carryon@1.2.0, commonlib@2.1.1, cooperativecombat@1.1.0, expandedfoods@1.6.6, fairplayguardian@1.1.2, hudclock@3.2.1, sailboat@1.1.0, naturalnight@2.0.0, nemi@1.1.0, petai@1.7.10, pixelwaypoints@1.0.0, claywheel@1.1.0, prospectorinfo@4.3.0, rhadditions@3.0.11, scarecrow@1.2.3, statushud@1.3.1, stonebakeoven@1.0.3, targetDotMedium@1.1.0, creative@1.18.6, vsquest@0.4.1, survival@1.18.6, waypointcolours@1.1.0, xlib@0.7.4, bullseye@2.5.0-rc.1, cats@1.7.0, doffanddonagain@3.0.0, em@2.2.1, playercorpse@1.7.1, rifteye@0.2.3, simplewinddirection@1.0.2, stonequarry@3.0.1, wolftaming@1.7.3, xinvtweaks@1.5.3, xskills@0.7.4, bricklayers@2.2.0
System.Exception: Don't know how to instantiate block of class 'BlockFirebrickKilnFlue' did you forget to register a mapping?
   at Vintagestory.Common.ClassRegistry.CreateBlock(String blockClass) in C:\Users\Tyron\Documents\vintagestory\game-net7\VintagestoryLib\Common\ClassRegistry.cs:line 129
   at Vintagestory.Common.BlockTypeNet.ReadBlockTypePacket(Packet_BlockType packet, IWorldAccessor world, ClassRegistry registry) in C:\Users\Tyron\Documents\vintagestory\game-net7\VintagestoryLib\Common\Network\BlockTypeNet.cs:line 399
   at Vintagestory.Client.NoObf.ClientSystemStartup.PopulateBlocks(Block[] blocks, Int32 start, Int32 maxCount) in C:\Users\Tyron\Documents\vintagestory\game-net7\VintagestoryLib\Client\Systems\Startup.cs:line 546
   at Vintagestory.Client.NoObf.ClientSystemStartup.LoadBlockTypes() in C:\Users\Tyron\Documents\vintagestory\game-net7\VintagestoryLib\Client\Systems\Startup.cs:line 363
   at Vintagestory.Client.NoObf.ClientSystemStartup.HandleServerAssets_Step1() in C:\Users\Tyron\Documents\vintagestory\game-net7\VintagestoryLib\Client\Systems\Startup.cs:line 338
   at Vintagestory.API.Common.TyronThreadPool.<>c__DisplayClass12_0.<QueueTask>b__0(Object a) in C:\Users\Tyron\Documents\vintagestory\game-net7\VintagestoryApi\Common\TyronThreadPool.cs:line 115
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 7/2/2023 9:45:03 AM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1880357656339671929, type 4
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.18.6.0
P3: 638f99ee
P4: openal32.dll
P5: 1.23.0.0
P6: 63dd31ad
P7: 40000015
P8: 00000000000df046
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER292E.tmp.WERInternalMetadata.xml

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_Vintagestory.exe_429d1884e4154ce84ac24791dbb411b784966a1e_a47ecedf_5be779a8-632e-4b75-97e4-edfc09902256

Analysis symbol: 
Rechecking for solution: 0
Report Id: 3095c553-bd99-4a3d-bcfa-4864f0ea7938
Report Status: 268566528
Hashed bucket: 0e6a3f4f1527bf7a2a185f536ccfeb79
Cab Guid: 0 }
--------------
{ TimeGenerated = 7/2/2023 9:45:02 AM, Site = , Source = Windows Error Reporting, Message = Fault bucket , type 0
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.18.6.0
P3: 638f99ee
P4: openal32.dll
P5: 1.23.0.0
P6: 63dd31ad
P7: 40000015
P8: 00000000000df046
P9: 
P10: 

Attached files:

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_Vintagestory.exe_429d1884e4154ce84ac24791dbb411b784966a1e_a47ecedf_5be779a8-632e-4b75-97e4-edfc09902256

Analysis symbol: 
Rechecking for solution: 0
Report Id: 3095c553-bd99-4a3d-bcfa-4864f0ea7938
Report Status: 131076
Hashed bucket: 
Cab Guid: 0 }
--------------
{ TimeGenerated = 7/2/2023 9:45:00 AM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.18.6.0, time stamp: 0x638f99ee
Faulting module name: openal32.dll, version: 1.23.0.0, time stamp: 0x63dd31ad
Exception code: 0x40000015
Fault offset: 0x00000000000df046
Faulting process id: 0x4408
Faulting application start time: 0x01d9acf36fe8c256
Faulting application path: C:\games\Vintagestory\Vintagestory.exe
Faulting module path: C:\games\Vintagestory\Lib\openal32.dll
Report Id: 3095c553-bd99-4a3d-bcfa-4864f0ea7938
Faulting package full name: 
Faulting package-relative application ID:  }```

**Comment by Craluminum2413** [comment]
Does it happen without any mods installed?

**Comment by Diometus** [comment]
> Does it happen without any mods installed?

Nope, but I play multiplayer and notice that if I disable the volumetric Shading (Updated) mod it goes away when playing in multiplayer, but if I keep it enabled and go into multiplayer, as I have said, I crash. Now if I leave the mod on for single-player with the same mods on the server I play multiplayer on, it also does not give me the error.

So I am uncertain if it's the mod volumetric Shading (Updated) itself or if it's something in the API because it works in single-player with the same mods but not in multiplayer.

I am unfamiliar with Vintage Story API, let alone the one for .NET 7 for v1.18.6 so I posted it here notating it is a problem occurring on 'modded'.  I did, however, ask the server owner on the multiplayer server I play on, and appears to be something new occurring and is also uncertain if its something being left out in the API by the mod I mentioned previously or if its something that is different in the multiplayer portion of Vintage Story's API versus single-player.

**Comment by Th3Dilli** [comment]
hey, I am not sure what is the culprit but the class that cannot be instantiated is not part of the main game so eventually, there may be some other issues when starting related to the mod that adds `BlockFirebrickKilnFlue`. maybe you can check your logs if this still happens and report that to the owner of that mod?

I know compatibility for net 7 with mods is not ideal right now but we hope things will be resolved with 1.18.8 which will be only net7 so mod authors can update their mods for net7 only

**Comment by sekelsta** [comment]
Vinternacht reports:
"I think I've finally narrowed down an old bug that people were encountering quite a bit for a while.  If you open the mod folder and update a mod from an older version to a newer version while the client is open (Not in a world, just open), it will throw an error on launch.  It doesn't crash, but it does say that two versions of that mod exist and won't work unless you restart your client, or manually go in and delete the cache folder.

The specific circumstances require the client to be open, and an old version of the mod removed and another one added without shutting down or restarting the server.  This isn't a problem in and of itself, except the very ability to click and open the mod folder implies hot-swapping while the client is running should be fine.

This specific series of events occurred on my machine, and on someone else's as well, so I believe it to be fairly repeatable."
